### PR TITLE
Add support for ``DataFrame.melt``

### PR DESCRIPTION
**Pull request by rjzamora**
Closes https://github.com/dask/dask-expr/issues/1002
dask/dask component: https://github.com/dask/dask/pull/11088

I'm not really sure if `melt` has been left out of the API for a specific reason. This PR adds basic support using `map_partitions`.

### CODE REVIEW

File: dask_expr/_collection.py [hunk 1]
```diff
@@ -3988,6 +3988,35 @@ def to_html(self, max_rows=5):
...
+    ):
+        meta = make_meta(
+            meta_nonempty(self._meta).melt(
+                id_vars=id_vars,
+                value_vars=value_vars,
+                var_name=var_name,
+                value_name=value_name,
+                col_level=col_level,
+            )
+        )
```
Comment (rjzamora) [inline dask_expr/_collection.py:4008]: Probably don't need this.

File: dask_expr/tests/test_collection.py [hunk 2]
```diff
@@ -2606,3 +2606,30 @@ def test_shape_integer(df):
...
+        {},
+        dict(id_vars="int"),
+        dict(value_vars="int"),
+        dict(value_vars=["obj", "int"], var_name="myvar"),
+        dict(id_vars="s1", value_vars=["obj", "int"], value_name="myval"),
+        dict(value_vars=["obj", "s1"]),
+        dict(value_vars=["s1", "s2"]),
+    ],
+)
+def test_melt(kwargs):
```
Comment (rjzamora) [inline dask_expr/tests/test_collection.py:2623]: Test copied from dask/dask for now.

File: dask_expr/_collection.py (hunk 1 above)
Comment (phofl) [reply dask_expr/_collection.py:4008]: Can you confirm this? Wouldn't want to merge it if we don't need it.

**Comment by rjzamora** [comment]
Thanks for the review @phofl ! Sorry - Should have marked this as a draft. I was curious why `melt` was never added, and didn't want to bury too much time in a PR only to find out that there are deeper issues I didn't know about. If the answer is: "non of the guiding benchmarks/workflows seemed to need it", then I'll be happy to clean this up and make test coverage is thorough both here and in dask/dask.

File: dask_expr/_collection.py (hunk 1 above)
Comment (rjzamora) [reply dask_expr/_collection.py]: Okay yeah. Default emulation works fine for the tests - Removing the unnecessary logic.

**Review by phofl — APPROVED** [review]
lgtm

can you add it to the api docs as well if not already done in your other PR?

**Comment by phofl** [comment]
thx

**Pull request merged**